File: Nivedita/Linkedlist/Add two numbers represented by linked lists.cpp
class Solution
{
    public:
    struct Node* reverse(struct Node* head){
        struct Node *prev = NULL, *curr = head, *next = NULL;
        while(curr){
            next = curr->next;
            curr->next = prev;
            prev = curr;
            curr = next;
        }
        return prev;
    }
    struct Node*add(struct Node*first,struct Node*second)
    {
        if(first==NULL)
    {
        return second;
    }
    if(second==NULL)
    {
        return first;
    }
    int sum=first->data+second->data;
    Node*p = new Node(sum%10);
    p->next=add(first->next,second->next);
    if(sum>=10)
    {
      p->next=add(p->next,new Node(1)); 
    }
    return p;
    }
    //Function to add two numbers represented by linked list.
    struct Node* addTwoLists(struct Node* first, struct Node* second)
    {
      struct Node*l1=reverse(first);
      struct Node*l2=reverse(second);
      return reverse(add(l1,l2));// code here
    }
};

File: Saummya/LINKEDLIST/GFG/Add two numbers represented by linked lists.py
class Solution:
    #Function to add two numbers represented by linked list.
    def reverse(self, head):
            prev = None
            temp = head
            while temp:
                nxt = temp.next
                temp.next = prev
                prev = temp
                temp = nxt
            head = prev
            
            return head
    
    
    
    def addTwoLists(self, first, second):
        # code here
        # return head of sum list
        sum1=0
        carry=0
        dummy=Node(-1)
        temp=dummy
        first = self.reverse(first)
        second = self.reverse(second)
        
        while(first is not None or second is not None or carry !=0):
            s = 0
            if first is not None:
                s += first.data
                first = first.next
            if second is not None:
                s += second.data
                second = second.next
                
            s += carry
            carry = s//10
            
            temp.next = Node(s%10)
            temp = temp.next
